### DISCUSSION TITLE
Maya Referencing USD File

### DISCUSSION THREAD

**Discussion by sshrestha-fractal** [Autodesk/maya-usd#1402] [Q&A]
Hi,

I noticed that referencing a USD file as a maya reference seems to work without any issues (at least I did not notice anything unusual) but maybe this route is not advisable? We are designing our usd workflow currently and storing models as alembic files that get usd-referenced in a container usd. In departments like rigging that need to have an editable version of the model in their files, we plan to reference the usd file as a maya reference (so in the anim shot files the usd model would be a nested reference within the maya rig reference).

I'm not sure if referencing usd file as a maya reference is a workflow supported by design or if its one of the by products of the maya-usd integration. Is this advisable or are there any pitfalls to this workflow? Any inputs or suggestions on how people are dealing with rigging workflows vis-a-vis usd would be super useful.

Thanks,
Sachin

**Comment by wtelford** [comment]
Referencing USD from a Maya file reference performs a data translation to the Maya data model. We have discussed whether this behavior is desirable or whether we would rather have it come in as a proxyShape. I'd like to investigate if this behavior could be configurable to give a pipeline a choice.

**Reply by sshrestha-fractal** [reply]
So, in the context of the use case I outlined above (in rigging), it would be preferable that the usd file gets translated as maya geometry during reference (which is the current behaviour). If I need to reference it as a non-editable geometry via usd proxy shape then I would certainly stay away from maya references but I can see having this as an option be useful in some cases.
